### ISSUE TITLE
chromatin accessibility task baseline - results reproduction

### ISSUE THREAD

**Issue by MidAtBest** [kundajelab/DART-Eval#8]
Dear authors,

Thank you very much for this benchmark it was very insightful.

I have a few questions regarding the `ChromBPNet` baseline used in task 4 i.e. the chromatin accessibility task.

## What I have understood from the paper

1. `ChromBPNet` was trained on Encode so in this case it is not truly ab-initio, right? 

> We utilized already trained ChromBPNet models from the ENCODE project for each dataset in this study 
(D.3 Ab initio Models)

2. The pretrained `ChromBPNet` was then finetuned on each chromatin accessibility dataset (i.e.finetuning), using both losses of the `ChromBPNet` (MSE + log-likelihood loss).

> The count prediction was trained using mean squared error loss, while the profile head was trained using log-likelihood loss  based on a multinomial distribution. Separate ChromBPNet models are trained on each chromatin accessibility dataset.
(D.3 Ab initio Models)

3. The DNALMs were finetuned using only the MSE loss (if so, has it been tried to finetune using the same approach as ChromBPNet ?)

## Reproducing your results

I would like to reproduce your results, I see that to do so I should run the following command line: 

```bash
python -m dnalm_bench.task_2_5_single.experiments.task_4_chromatin_activity.eval_ab_initio.chrombpnet_baseline $CELL_TYPE $CHROMBPNET_MODEL_FILENAME
```

However, I am encountering some issues doing so:

1. The folder https://www.synapse.org/Synapse:syn60581108 does not contain any ChromBPNet checkpoint, and I cannot seem to find these weights (or even predictions) anywhere. Would you have the weights available in order to reproduce these results ?

2. In the evaluation file I see that the following is used `chroms_test = chroms_train + chroms_val + chroms_test`, is this done in practice ? Are you comparing the results on the test set for the DNALMs to the result on the 3 splits for `ChromBPNet` ? 

I would like to, once again, thank you for the work you've done on this paper and would be extremely grateful for any answers you might have.

**Comment by austintwang** [comment]
Hi, these are great questions!

> ChromBPNet was trained on Encode so in this case it is not truly ab-initio, right?

> The pretrained ChromBPNet was then finetuned on each chromatin accessibility dataset (i.e.finetuning), using both losses of the ChromBPNet (MSE + log-likelihood loss).

ChromBPNet isn't a multi-task model like Enformer or Borzoi. Instead, ChromBPNet models are individually trained on each dataset. The datasets we use for task 4 in DART-Eval are part of ENCODE, so we just use ChromBPNet models trained on the corresponding datasets.

> The DNALMs were finetuned using only the MSE loss (if so, has it been tried to finetune using the same approach as ChromBPNet ?)

We haven't looked into that, and it would be interesting to see if anything changes. The practical challenge would be reconciling the base-pair-resolution output with the various tokenization schemes these DNALMs use. It may not make a huge difference since the profile log-likelihood objective is orthogonal to the total counts MSE objective, though we do find for BPNet/ChromBPNet modeling the profile helps significantly with interpretability. 

For the reproducibility questions, let me check with the co-authors (@ArpiS @amanpatel101) and we'll get back to you soon. We've been running into some storage limits for Synapse, but we'll try and find an alternate way of getting the checkpoints to you.

Hope this helps! Let us know if you have any other questions.

**Comment by amanpatel101** [comment]
Hi! Regarding reproducibility, your comment made us realize that we had pushed the wrong version of the chrombpnet_baseline.py file (which contained some internal experiments) to the repo. It has been corrected now, and you'll see everything matches up with the rest of the models. Sorry about that!

The ChromBPNet models were not trained as part of this study, but they are publicly available from the ENCODE consortium. You can go to encodeproject.org and search for the relevant dataset, which can be found in Table S1 of our paper. Click on the experiment associated with that dataset, and then click on the link next to "Annotation (ChromBPNet-model)". Finally, you can download the tar file containing the models. We specifically used the fold 3 models in our study. Let us know if you have any trouble finding the models!

**Comment by MidAtBest** [comment]
Thank you so much for the additional information, I have a much better understanding now.
I believe I have found all that I need on my end. 

Best regards,
Shawn
